feature(groups): group avatars now use serve-file handler

hypeJunction opened the pull request:
Group avatars are now served with elgg_get_inline_url()

hypeJunction committed:
feature(groups): group avatars now use serve-file handler

Group avatars are now served with elgg_get_inline_url()

hypeJunction commented:
@jdalsem @jeabakker Could you check this works with all of your custom handlers? I don't expect any problems, but would be nice to double check.

mrclay commented:
No hooks/resource view to worry about here :+1:

jeabakker commented on mod/groups/icon.php:
```diff
@@ -1,57 +1,26 @@
-	die('This file cannot be called directly.');
-}
+$guid = get_input('group_guid');
+$size = get_input('size') ? : 'medium';
```
why not 
```php
$size = get_input('size', 'medium');
```

hypeJunction replied:
`?size=` would give an empty string and not default to `medium` AFAIK

jeabakker replied:
true (i think), but if i for some stupid reason have a custom size ```0``` i could never get it ;)
what would be the best way to solve this?

hypeJunction replied:
Lol. Just don't :)

hypeJunction replied:
```php
$size = get_input('size');
if (!isset($size) || $size === '') {
   $size = 'medium';
}
```

jeabakker replied:
I love it!

jeabakker commented:
I think this looks good

hypeJunction commented:
@jeabakker Updated size variable handling
Updated deprecation messages to 2.2

mrclay commented:
Should mention deprecations in docs/guides/upgrading (until there's a better place).

hypeJunction committed:
deprecate(groups): new file service deprecated groupicon page handler

groupicon page handler (groups_icon_handler()) is deprecated along with
/mod/groups/icon.php file

hypeJunction commented:
@mrclay Adding a second `deprecate` commit that updates the docs
